replication.proto

```diff
@@ -0,0 +1,155 @@
+
+syntax = "proto3";
+package replication;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = ".;replication";
+
+extend google.protobuf.FieldOptions {
+  // Indicates that a field MAY contain information that is sensitive
+  // and MUST be treated as such (e.g. not logged).
+  bool replication_secret = 1059;
+}
+
+// Controller holds the RPC Methods for replication and all the methods it
+// exposes should be idempotent.
+service Controller {
+  // EnableVolumeReplication RPC call to enable the volume replication.
+  rpc EnableVolumeReplication (EnableVolumeReplicationRequest)
+  returns (EnableVolumeReplicationResponse) {}
+  // DisableVolumeReplication RPC call to disable the volume replication.
+  rpc DisableVolumeReplication (DisableVolumeReplicationRequest)
+  returns (DisableVolumeReplicationResponse) {}
+  // PromoteVolume RPC call to promote the volume.
+  rpc PromoteVolume (PromoteVolumeRequest)
+  returns (PromoteVolumeResponse) {}
+  // DemoteVolume RPC call to demote the volume.
+  rpc DemoteVolume (DemoteVolumeRequest)
+  returns (DemoteVolumeResponse) {}
+  // ResyncVolume RPC call to resync the volume.
+  rpc ResyncVolume (ResyncVolumeRequest)
+  returns (ResyncVolumeResponse) {}
+}
+
+// EnableVolumeReplicationRequest holds the required information to enable
+// replication on a volume.
+message EnableVolumeReplicationRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
```

Comment:
What is `CO`?

Reply:
https://hackmd.io/1Pbj_9rzQ2SwK80GK6-qWg#Terminology

Reply:
It isn't obvious from the comment (without a link to HackMD doc).

Reply:
We still need to add a readme for complete spec and operations, will cover all details in that